contrib/scripts/check-api-code-gen.sh

```diff
@@ -38,7 +38,7 @@ make generate-hubble-api
 git add --intent-to-add .
 
 # Check for diff
-diff="$(git diff)"
+diff="$(git diff --staged)"
```

Comment:
I think this breaks the detection of new files, again, though. Quick test repo, `existing` is a committed file, I've run `rm existing; echo "blub2" > new` and then `git add --intent-to-add .` . `git diff` shows the diff in `new`, `git diff --staged` shows the diff in `existing`, `git diff` shows the diff in `new`...

```
david@lynx:~/tmprepo:main -$ git add --intent-to-add .
david@lynx:~/tmprepo:main -$ git diff --staged        
diff --git a/existing b/existing
deleted file mode 100644
index 40142d0..0000000
--- a/existing
+++ /dev/null
@@ -1 +0,0 @@
-blub
david@lynx:~/tmprepo:main -$ git diff         
diff --git a/new b/new
new file mode 100644
index 0000000..4c30ff2
--- /dev/null
+++ b/new
@@ -0,0 +1 @@
+blub2
```

Reply:
why using `--intent-to-add` at all? (only tested it with a normal `git add .` :man_facepalming: - then it works with `git diff --staged` for new files too (at least on my machine :P ))